src/vs/workbench/contrib/inlineChat/browser/inlineChatSessionServiceImpl.ts

```diff
@@ -62,7 +64,9 @@ export class InlineChatSessionServiceImpl implements IInlineChatSessionService {
 		@ILogService private readonly _logService: ILogService,
 		@IInstantiationService private readonly _instaService: IInstantiationService,
 		@IEditorService private readonly _editorService: IEditorService,
-	) { }
+	) {
+		this.onDidChangeSessionEnablementStatus = this._inlineChatService.onDidChangeProvidersEnablementStatus;
```

Comment:
This feels wrong should I just depend on the inlineChatService in the controller?